src/overlays/actors/ovl_Bg_Danpei_Movebg/z_bg_danpei_movebg.c

```diff
@@ -27,38 +36,182 @@ ActorInit Bg_Danpei_Movebg_InitVars = {
+
+    if (SubS_IsObjectLoaded(this->objectSlot, play)) {
+        gSegments[6] = OS_K0_TO_PHYSICAL(play->objectCtx.slots[this->objectSlot].segment);
+        this->dyna.actor.objectSlot = this->objectSlot;
+        this->dyna.actor.draw = func_80AF74CC;
+        Actor_ProcessInitChain(thisx, sInitChain);
+        DynaPolyActor_LoadMesh(play, &this->dyna, D_80AF7538[BGDANPEIMOVEBG_GET_TYPE(thisx)]);
+        this->dList = D_80AF7534[BGDANPEIMOVEBG_GET_TYPE(thisx)];
+
+        if (BGDANPEIMOVEBG_GET_TYPE(thisx) == 0) {
+            this->dyna.actor.gravity = 0.0f;
+            this->dyna.actor.speed = 1.0f;
+            SubS_ActorPathing_Init(play, &thisx->world.pos, thisx, &this->actorPath, play->setupPathList,
+                                   BGDANPEIMOVEBG_GET_PATH_INDEX(thisx), 0, 0, 0, 0);
+            this->actionFunc = func_80AF70FC;
+        }
+    }
+}
+
+void BgDanpeiMovebg_Destroy(Actor* thisx, PlayState* play) {
+    BgDanpeiMovebg* this = THIS;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void BgDanpeiMovebg_Update(Actor* thisx, PlayState* play) {
+    BgDanpeiMovebg* this = THIS;
+
+    this->actionFunc(this, play);
+    this->prevFlags = this->flags;
+    this->flags &= ~1;
+}
+
+u16 func_80AF705C(BgDanpeiMovebg* this, u16 flags) {
+    u16 newFlags = flags & ~0x1C;
+
+    if (DynaPolyActor_IsActorOnTop(&this->dyna) && (this->flags & 1)) {
+        newFlags |= 4;
+    }
+    if (DynaPolyActor_IsPlayerOnTop(&this->dyna)) {
+        if (gSaveContext.save.playerForm == PLAYER_FORM_DEKU) {
+            newFlags |= 8;
+        } else {
+            newFlags |= 0x10;
+        }
+    }
+    return newFlags;
+}
+
+void func_80AF70FC(BgDanpeiMovebg* this, PlayState* play) {
+    this->flags = func_80AF705C(this, this->flags);
+    if (this->flags & 0xC) {
+        if (this->flags & 4) {
+            this->flags |= 2;
+        }
+        this->actionFunc = func_80AF71FC;
+    } else if (this->flags & 0x10) {
+        if (!(this->prevFlags & 0x10)) {
+            Actor_PlaySfx(&this->dyna.actor, NA_SE_EV_SINK_WOOD_FLOOR);
+        }
+        Math_SmoothStepToF(&this->dyna.actor.world.pos.y, this->dyna.actor.home.pos.y - 8.0f, 1.0f, 2.0f, 0.01f);
+    } else {
+        if (this->prevFlags & 0x10) {
+            Actor_PlaySfx(&this->dyna.actor, NA_SE_EV_REBOUND_WOOD_FLOOR);
+        }
+        Math_SmoothStepToF(&this->dyna.actor.world.pos.y, this->dyna.actor.home.pos.y, 1.0f, 2.0f, 0.01f);
+    }
+}
+
+void func_80AF71FC(BgDanpeiMovebg* this, PlayState* play) {
+    SubS_ActorPathing_Update(play, &this->actorPath, SubS_ActorPathing_ComputePointInfo, func_80AF6DE0,
+                             SubS_ActorPathing_MoveWithoutGravityReverse, SubS_ActorPathing_SetNextPoint);
+    Actor_PlaySfx_Flagged(&this->dyna.actor, NA_SE_EV_PLATE_LIFT_LEVEL - SFX_FLAG);
+    if (this->actorPath.flags & ACTOR_PATHING_REACHED_END_TEMPORARY) {
+        if (this->flags & 2) {
+            this->flags &= ~0x2;
```

Comment:
Maybe something like this? Just to be more consistent
```suggestion
        if (this->flags & 2) {
            this->flags &= ~2;
```